pkg/api/oidc_login_handler.go

```diff
@@ -38,6 +39,16 @@ func NewOIDCLoginPageHandler(sessionStore sessions.Store, oauthConfig *oauth2.Co
 			writeError(w, http.StatusInternalServerError, "Failed to save auth session")
 			return
 		}
-		http.Redirect(w, r, oauthConfig.AuthCodeURL(state), http.StatusTemporaryRedirect)
+		opts := make([]oauth2.AuthCodeOption, 0, len(r.URL.Query()))
+		for k, v := range r.URL.Query() {
+			if len(v) < 1 {
+				continue
+			}
+			opts = append(opts, oauth2.SetAuthURLParam(k, v[0]))
```

Comment:
I don't know enough about OIDC, but this scares me: the user supplies parameters that we pass directly to the OIDC server without understanding what they do.  We will override _some_ of these on l. 49-41... but do we know that we override _everything_ required?

I would feel safer if we only allowed whitelisted options here.

Reply:
After f2fing I understand that these parameters go (_only_) into the redirection URL returned from the OIDC provider.  Since there is nothing secret about the generated URL, there is no attack vector here.